src/hooks/useUserOnboardingContext.ts

```diff
@@ -15,54 +15,99 @@ limitations under the License.
 */
 
 import { logger } from "matrix-js-sdk/src/logger";
-import { ClientEvent, IMyDevice, Room } from "matrix-js-sdk/src/matrix";
-import { useCallback, useEffect, useState } from "react";
+import { ClientEvent } from "matrix-js-sdk/src/matrix";
+import { useCallback, useEffect, useRef, useState } from "react";
 
 import { MatrixClientPeg } from "../MatrixClientPeg";
+import { Notifier } from "../Notifier";
 import DMRoomMap from "../utils/DMRoomMap";
-import { useEventEmitter } from "./useEventEmitter";
 
 export interface UserOnboardingContext {
-    avatar: string | null;
-    myDevice: string;
-    devices: IMyDevice[];
-    dmRooms: {[userId: string]: Room};
+    hasAvatar: boolean;
+    hasDevices: boolean;
+    hasDmRooms: boolean;
+    hasNotificationsEnabled: boolean;
+}
+
+const USER_ONBOARDING_CONTEXT_INTERVAL = 5000;
+
+function useRefOf<T extends [], R>(value: (...values: T) => R): (...values: T) => R {
+    const ref = useRef(value);
+    ref.current = value;
+    return useCallback(
+        (...values: T) => ref.current(...values),
+        [],
+    );
```

Comment:
I'm completely failing to wrap my head around what this & the handler below are doing - it's taking a function, wrapping it in a Ref, then returns the same function but using the ref-wrapped function? But then the handler doesn't take any arguments? Is this a way of memoising the function?

Reply:
It’s a way to obtain a stable handle to a function that remains the same even if the actual function changes. This way the handler, which is memoized, won’t be updated even if the actual callback here changes.

I saw this idea in useEventEmitter and loved it, but I should definitely comment it better, yeah.